gensim/summarization/bm25.py

```diff
@@ -162,6 +162,48 @@ def get_scores(self, document):
         scores = [self.get_score(document, index) for index in range(self.corpus_size)]
         return scores
 
+    def get_scores_bow(self, document):
+        """Computes and returns BM25 scores of given `document` in relation to
+        every item in corpus.
+
+        Parameters
+        ----------
+        document : list of str
+            Document to be scored.
+
+        Returns
+        -------
+        list of float
+            BM25 scores.
+
+        """
+        scores = []
+        for index in range(self.corpus_size):
+            score = self.get_score(document, index)
+            if score > 0:
```

Comment:
In that case `len(scores) <= self.corpus_size`, why?

Reply:
Because it's actually quite sparse array, isn't it?
In `summarizer._set_graph_edge_weights` such documents with little weight will be dropped anyway, so there is no reason to waste extra memory. And what's more, if we are needed a dense array we can uncompactify this bow.

Reply:
I don't get, how we understand ids of documents that have 0 scores in that case?